### PR TITLE
PS-1726 Extract workspace providers to a separate library

### PR DESCRIPTION
**Pull request by joker806**
https://keboola.atlassian.net/browse/PS-1726

### CODE REVIEW

**Comment by odinuv** [comment]
Hřeje mě to u srdíčka když vidím jak ten kód mizí :)

File: composer.json [hunk 1]
```diff
@@ -11,18 +11,19 @@
...
-        "keboola/input-mapping": "^12.2",
         "keboola/php-utils": "^2.0",
+        "keboola/storage-api-php-client-branch-wrapper": "^1.0.0",
         "keboola/syrup": "^11.3",
+        "keboola/workspace-provider": "^1.0",
         "mustache/mustache": "^2.13",
         "symfony/validator": "^v2.8.52|^v4.4.11",
-        "vkartaviy/retry": "^v0.2.2",
-        "ext-zip": "*"
+        "vkartaviy/retry": "^v0.2.2"
```
Comment (pivnicek) [inline composer.json:26]: not relevant here, but we should probably switch to our fork of this eventually https://github.com/keboola/retry

**Review by pivnicek — APPROVED** [review]
image
👍 
🤞 for tests :-)

File: composer.json (hunk 1 above)
Comment (odinuv) [reply composer.json:26]: yes, but not relevant here :D

Comment (joker806) [reply composer.json:26]: testy pohoda.. I have removed all the related tests :D

**Review by odinuv — CHANGES_REQUESTED** [review]
👍

**Review by odinuv — APPROVED** [review]

**Pull request merged**